build.sh

```diff
@@ -0,0 +1,119 @@
+#!/bin/bash
+
+set -Eeuo pipefail
+
+trap 'echo "Build failed."' ERR
+
+usage() {
+  echo "${0} CMD"
+  echo "Available CMDs:"
+  echo -e "\todroid_h4              - build Dasharo compatible with Hardkernel ODROID H4"
+  echo -e "\tqemu                   - build Dasharo compatible with QEMU Q35"
+}
+
+INPUT_IMAGE="${INPUT_IMAGE:-image.rom}"
+
+build_odroid_h4() {
+  local flags="-D CRYPTO_PROTOCOL_SUPPORT=TRUE -D SIO_BUS_ENABLE=TRUE \
+    -D PERFORMANCE_MEASUREMENT_ENABLE=TRUE \
+    -D MULTIPLE_DEBUG_PORT_SUPPORT=TRUE -D BOOTSPLASH_IMAGE=TRUE \
+    -D BOOT_MANAGER_ESCAPE=TRUE"
+  if [ ! -f "$INPUT_IMAGE" ]; then
+    echo "$INPUT_IMAGE doesn't exist or isn't file."
+    echo "You can set different path to vendor image in INPUT_IMAGE variable"
+    exit 1
+  fi
+  docker build -t sbl --network=host .
+  build_edk2 "edk2-stable202505" "$flags"
+  build_slimloader odroidh4 "0xAAFFFF0C"
+  echo "Result binary placed in $PWD/Outputs/odroidh4/ifwi-release.bin"
+  sha256sum Outputs/odroidh4/ifwi-release.bin >Outputs/odroidh4/ifwi-release.bin.sha256
+}
+
+build_qemu() {
+  docker build -t sbl --network=host .
+  build_slimloader qemu ""
+  echo "Result binary placed in $PWD/Outputs/qemu/SlimBootloader.bin"
+  sha256sum Outputs/qemu/SlimBootloader.bin >Outputs/qemu/SlimBootloader.bin.sha256
+}
+
+build_edk2() {
+  EDK2_VERSION="$1"
+  FLAGS="$2"
+  rm -rf edk2
+  mkdir edk2
+  cd edk2
+  # clone one commit only
+  git init
+  git remote remove origin 2>/dev/null || true
+  git remote add origin https://github.com/tianocore/edk2.git
+  git fetch --depth 1 origin "$EDK2_VERSION"
+  git checkout FETCH_HEAD --force
+  git submodule update --init --checkout --recursive --depth 1
+
+  docker run --rm -i -u "$UID" -v "$PWD":/edk2 -w /edk2 sbl /bin/bash <<EOF
+    source edksetup.sh
+    make -C BaseTools
+    python ./UefiPayloadPkg/UniversalPayloadBuild.py -t GCC5 -o Dasharo -b RELEASE \
+      $FLAGS
+EOF
+  cd ..
+}
+
+build_slimloader() {
+  platform="$1"
+  platform_data="$2"
+  input="$(realpath "$INPUT_IMAGE")"
+  git submodule update --init --checkout --recursive --depth 1
+  if [ "$platform" = "qemu" ]; then
+    docker run --rm -i -u $UID -v "$PWD":/home/docker/slimbootloader \
+      -w /home/docker/slimbootloader sbl /bin/bash <<EOF
+        set -e
+        export SBL_KEY_DIR="\${PWD}/SblKeys"
```

Comment:
I would let the SBL_KEY_DIR variable be defined by the script caller.

1. We need SblTestKeys inside the repo so that build can be reproduces (aka dev_signed builds). The script may set SBL_KEY_DIR to default to `${sbl_root}/SblTestKeys` if the variable is not set.
2. If we need production signed builds, the caller exports or gives SBL_KEY_DIR when calling. These keys will be protected externally and only used when building a release image.

Reply:
Ad.2. Done.